### PR TITLE
Return metadata from decrypt function

### PR DESCRIPTION
**Pull request by ThibaudDauce**

### CODE REVIEW

File: src/interfaces/wasm_bindgen/tests.rs [hunk 1]
```diff
@@ -175,9 +175,17 @@ fn test_encrypt_decrypt() {
         res,
         Uint8Array::from(authentication_data.as_slice()),
     )
-    .unwrap();
+    .unwrap()
+    .to_vec();
+
+    let mut bytes = res.as_slice();
+
+    let additional_data_length = leb128::read::unsigned(&mut bytes).unwrap() as usize;
```
Comment (tbrezot) [inline src/interfaces/wasm_bindgen/tests.rs:183]: Use a deserializer here:
``` rust
let de = Deserializer::new(res.as_slice());
let additional_data = de.read_vec()?;
let cleartext = de.finalize();
```

File: src/interfaces/wasm_bindgen/hybrid_cc_aes.rs [hunk 2]
```diff
@@ -294,5 +299,17 @@ pub fn webassembly_hybrid_decrypt(
         )
         .map_err(|e| JsValue::from_str(&format!("Error decrypting ciphertext: {e}")))?;
 
-    Ok(Uint8Array::from(cleartext.as_slice()))
+    let mut result = vec![];
```
Comment (tbrezot) [inline src/interfaces/wasm_bindgen/hybrid_cc_aes.rs:302]: You should use the serializer from CryptoCore. Something like:
``` rust
let ser = Serializer::new();
ser.write_vec(cleartext_header.additional_data.as_slice())?;
ser.write_array(cleartext.as_slice())?;
Ok(Uint8Array::from(ser.value()))
```

**Review by tbrezot — CHANGES_REQUESTED** [review]

File: src/interfaces/wasm_bindgen/hybrid_cc_aes.rs (hunk 2 above)
Comment (ThibaudDauce) [reply src/interfaces/wasm_bindgen/hybrid_cc_aes.rs]: Done in 2937cfe

File: src/interfaces/wasm_bindgen/tests.rs (hunk 1 above)
Comment (ThibaudDauce) [reply src/interfaces/wasm_bindgen/tests.rs]: Done in 2937cfe

**Review by tbrezot — APPROVED** [review]

**Pull request merged**